### ISSUE TITLE
Link fails on FreeBSD, ambiguous s64 type

### ISSUE THREAD

**Issue by AMDmi3** [minetest/minetest#9063]
##### Minetest version
```
5.1.0
```

##### Summary
5.1.0 fails to link on FreeBSD:
```
src/CMakeFiles/minetest.dir/main.cpp.o: In function `main':
main.cpp:(.text+0x29f8): undefined reference to `FileLogOutput::setFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, long long)'
c++: error: linker command failed with exit code 1 (use -v to see invocation)
```

The cause is s64 type, which has different meanings in `log.cpp`, and in `main.cpp`, so `setFile` signatures are different at definition and when it's called:

```
% nm -C log.cpp.o | grep setFile
00000000000014b0 T FileLogOutput::setFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, long)
% nm -C main.cpp.o | grep setFile
                 U FileLogOutput::setFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, long long)
```

Still investigating where different types are defined and why there's no compiler error.

##### Steps to reproduce
Try to build on FreeBSD

**Comment by SmallJoker** [comment]
I suspect the macOS build suffers from the same problem (#9056).

**Comment by AMDmi3** [comment]
> I suspect the macOS build suffers from the same problem

Yep, it seems so.

Update: both conflicting s64 definitions come from irrlicht's `irrTypes.h`, but from different lines:

```
#if __WORDSIZE == 64
typedef long int                s64;
#else
__extension__ typedef long long         s64;
#endif
```

`__WORDSIZE` is defined in `stdint.h`, and, thus, `cstdint`.

`cstdint` is included from `irrlichttypes.h`. There's comment on this issue:

https://github.com/minetest/minetest/blob/master/src/irrlichttypes.h#L22-L23

I guess this should be fixed by including `irrlichttypes.h` in `main.cpp`.

**Comment by AMDmi3** [comment]
Fix confirmed, see #9064.

**Comment by SmallJoker** [comment]
#9064 merged
